EngineLayer/SpectralLibrarySearch/SpectralLibrary.cs

```diff
@@ -330,6 +351,120 @@ private LibrarySpectrum ReadLibrarySpectrum(StreamReader reader, bool onlyReadHe
                     //TODO: figure out terminus
                     FragmentationTerminus terminus = (FragmentationTerminus)Enum.Parse(typeof(FragmentationTerminus), "None", true);
 
+                    //TODO: figure out amino acid position
+                    var product = new Product(peakProductType, terminus, experMz, fragmentNumber, 0, neutralLoss);
+
+                    matchedFragmentIons.Add(new MatchedFragmentIon(ref product, experMz, experIntensity, fragmentCharge));
+                }
+            }
+
+            return new LibrarySpectrum(sequence, precursorMz, z, matchedFragmentIons, rt);
+        }
+
+        private LibrarySpectrum ReadLibrarySpectrum_pDeep(StreamReader reader, bool onlyReadHeader = false)
```

Comment:
ideally you would not have a separate method that's a near duplicate of `ReadLibrarySpectrum` but i'll leave that up to you. maybe the pdeep library is too different to adapt ReadLibrarySpectrum. but it's not ideal to have this separate nearly-identical code